### PR TITLE
fix: multiple deposits of unknown brand

### LINKED ISSUE #6961: smart wallet fails on 2nd payment from unknown brand
## Describe the bug

Payments bearing unknown brands are supposed to be queued until the corresponding issuer is added. In the queue implementation, an array is `harden`ed and then we later `.push()` to it, which throws.

## To Reproduce

Run this test:
https://github.com/Agoric/agoric-sdk/blob/b63db15156c2cde188020c9a8fe8f90eb26e1da2/packages/inter-protocol/test/smartWallet/test-psm-integration.js#L363

## Expected behavior

doesn't throw. Payment is deposited when the issuer comes along

## Additional context

Unknown brands are only possible in MN-2 / MN-3 / devnet.

cc @turadg

### PR DESCRIPTION
**Pull request by turadg**
closes: #6961

## Description

Makes a new array instead of try to push to the hardened one. Also improved docs on how unknown brand deposits are handled.

### Testing Considerations

Failing test now passing.

### CODE REVIEW

File: packages/smart-wallet/src/smartWallet.js [hunk 1]
```diff
@@ -374,11 +374,13 @@ export const prepareSmartWallet = (baggage, shared) => {
             return E(invitationPurse).deposit(payment);
           }
 
-          // When there is no purse, queue the payment
+          // When there is no purse, save the payment into a queue.
+          // It's not yet ever read but a future version of the contract can
```
Comment (dckc) [inline packages/smart-wallet/src/smartWallet.js:378]: Interesting... so we don't yet trigger deposits when new issuers come along, but we support upgrade to a wallet that does.

Hm... is there a way for somebody to send an ERTP payment with a brand not known to the vbank? I can't think of one. I was thinking we might owe an issue for the case of newly-approved tokens, but I think we're internally consistent as-is.

File: packages/smart-wallet/src/smartWallet.js [hunk 2]
```diff
@@ -374,11 +374,13 @@ export const prepareSmartWallet = (baggage, shared) => {
...
           if (queues.has(brand)) {
-            queues.get(brand).push(payment);
+            const extant = queues.get(brand);
+            queues.set(brand, harden([...extant, payment]));
```
Comment (dckc) [inline packages/smart-wallet/src/smartWallet.js:381]: this seems to be the gist of it.

**Review by dckc — APPROVED** [review]
interesting approach.

File: packages/smart-wallet/src/smartWallet.js (hunk 1 above)
Comment (turadg) [reply packages/smart-wallet/src/smartWallet.js:378]: I think that's right

**Review by dckc — APPROVED** [review]

**Pull request merged**